cloudvolume/threaded_queue.py

```diff
@@ -175,30 +179,62 @@ def _consume_queue_execution(self, fn):
         self.processed += 1
         self._queue.task_done()
 
-  def wait(self):
+  def _check_errors(self):
+    try:
+      err = self._error_queue.get(block=False) 
+      self._error_queue.task_done()
+      self.kill_threads()
+      raise err
+    except Queue.Empty:
+      pass
+
+  def wait(self, progress=None):
     """
     Allow background threads to process until the
     task queue is empty. If there are no threads,
     in theory the queue should always be empty
     as processing happens immediately on the main thread.
 
-    Required: None
+    Optional:
+      progress: (bool or str) show a tqdm progress bar optionally
+        with a description if a string is provided
     
     Returns: self (for chaining)
 
     Raises: The first exception recieved from threads
     """
-    if len(self._threads):
-        self._queue.join()
+    if not len(self._threads):
+      return self
 
-    try:
-      # no blocking because we're guaranteed
-      # that all threads have finished processing
-      err = self._error_queue.get(block=False) 
-      self._error_queue.task_done()
-      raise err
-    except Queue.Empty:
-      pass
+    desc = None
+    if type(progress) is str:
+      desc = progress
+
+    last = self._inserted
+    with tqdm(total=self._inserted, disable=(not progress), desc=desc) as pbar:
+      # Allow queue to consume, but check up on
+      # progress and errors every tenth of a second
+      while not self._queue.empty():
+        size = self._queue.qsize()
+        delta = last - size
+        if delta != 0: # We should crash on negative numbers
```

Comment:
how is it possible to get negative numbers?

Reply:
Programming errors when you are changing the code.

Reply:
Also if someone interferes with self._inserted